File: 30days/15/15.rb
class Node
    attr_accessor :data, :next

    def initialize(data)
        @data = data
        @next = nil
    end
end

class Solution
    def insert(head, value)
        if head.nil?
            return Node.new(value)
        elsif head.next.nil?
            head.next = Node.new(value)
        else
            insert(head.next, value)
        end
        head
    end

    def display(head)
        current = head
        while current
            print current.data, " "
            current = current.next
        end
    end
end

mylist = Solution.new
head = nil
T = gets.to_i
for i in 1..T
    data = gets.to_i
    head = mylist.insert(head, data)
end
mylist.display(head)

File: 30days/15/15.php
<?php
class Node{
    public $data;
    public $next;
    function __construct($d)
    {
        $this->data = $d;
        $this->next = NULL;
    }
}

class Solution{
    function insert($head,$data) {
        if ($head === null) {
            return new Node($data);
        } else {
            $tmp = $head;
            while ($tmp->next !== null) {
                $tmp = $tmp->next;
            }
            $tmp->next = new Node($data);
            return $head;
        }
    }

    function display($head){
        $start=$head;
        while($start){
            echo $start->data,' ';
            $start=$start->next;
        }
    }
}
$T=intval(fgets(STDIN));
$head=null;
$mylist=new Solution();
while($T-->0){
    $data=intval(fgets(STDIN));
    $head=$mylist->insert($head,$data);
}
$mylist->display($head);
?>
